### PR TITLE
[Config] Reasonable default settings for ARC

### PR DESCRIPTION
**Pull request by klausenbusk**
Fix: #2703

---
cc @prehor

I don't think `allow_username_mismatch = true` is needed as we only sign inbound mails. `allow_username_mismatch` only affects authenticated users as far as I understand.

### CODE REVIEW

File: conf/modules.d/arc.conf [hunk 1]
```diff
@@ -26,25 +26,27 @@ arc {
   # If false, messages with empty envelope from are not signed
   allow_envfrom_empty = true;
   # If true, envelope/header domain mismatch is ignored
-  allow_hdrfrom_mismatch = false;
+  allow_hdrfrom_mismatch = true;
   # If true, multiple from headers are allowed (but only first is used)
-  allow_hdrfrom_multiple = false;
+  allow_hdrfrom_multiple = true;
```
Comment (vstakhov) [inline conf/modules.d/arc.conf]: Why?

**Comment by klausenbusk** [comment]
You correct, that doesn't make sense. I misinterpreted the meaning of that 
option.


I will fix it when I get home.

File: conf/modules.d/arc.conf (hunk 1 above)
Comment (klausenbusk) [reply conf/modules.d/arc.conf]: You correct, that doesn't make sense. I misinterpreted the meaning of that
option.

Fixed.

File: conf/modules.d/arc.conf [hunk 2]
```diff
@@ -26,25 +26,27 @@ arc {
...
   # If true, envelope/header domain mismatch is ignored
-  allow_hdrfrom_mismatch = false;
+  allow_hdrfrom_mismatch = true;
   # If true, multiple from headers are allowed (but only first is used)
   allow_hdrfrom_multiple = false;
   # If true, username does not need to contain matching domain
   allow_username_mismatch = false;
   # If false, messages from authenticated users are not selected for signing
-  auth_only = true;
+  auth_only = false;
```
Comment (klausenbusk) [inline conf/modules.d/arc.conf]: I don't think this make any difference as we don't know if the mail is from a authenticated user (ex: `user1@example.com` -> `user2@example.com`),at least that what my testing shows.

Comment (klausenbusk) [reply conf/modules.d/arc.conf]: Never mind, it was a configuration error.

**Comment by whataboutpereira** [comment]
These defaults seem effective in getting incoming mail ARC signed. I do hope the config will get reasonable defaults.

**Comment by prehor** [comment]
> I don't think `allow_username_mismatch = true` is needed as we only sign inbound mails. `allow_username_mismatch` only affects authenticated users as far as I understand.

You are right, it is not necessary. I use it in `dkim_signing.conf` and blindly copied it to `arc.conf`.

**Comment by andryyy** [comment]
What do you think, @vstakhov

**Pull request merged**